test/model/src/driver/kv_workload.cpp

```diff
@@ -37,6 +37,168 @@ extern "C" {
 
 namespace model {
 
+namespace operation {
+
+/*
+ * parse --
+ *     Parse an operation from a string. Throw an exception on error.
+ */
+any
+parse(const char *str)
+{
+    const char *p = str;
+
+    /* Get the operation name. */
+    std::string name;
+    while (*p != '\0' && (std::isalnum(*p) || *p == '_'))
+        name += *(p++);
+
+    while (*p != '\0' && std::isspace(*p))
+        p++;
+    if (*(p++) != '(')
+        throw model_exception("Expected '('");
+
+    /* Get the arguments. */
+    std::vector<std::string> args;
+    std::string current;
+    bool done = false;
+    bool escaped = false;
+    bool had_quotes = false;
+    bool quotes = false;
+    while (*p != '\0') {
+        char c = *(p++);
+
+        if (escaped) {
+            current += c;
+            escaped = false;
+            continue;
+        }
+
+        if (quotes) {
+            if (c == '\"') {
+                quotes = false;
+                continue;
+            }
+            if (c == '\\') {
+                escaped = true;
+                continue;
+            }
+            current += c;
+            continue;
+        }
+
+        if (std::isspace(c))
+            continue;
+
+        if (c == '\"') {
+            quotes = true;
+            had_quotes = true;
+            continue;
+        }
+
+        if (c == ',' || c == ')') {
+            if (!(c == ')' && current.empty() && !had_quotes))
```

Comment:
Add comment before line 100
/* if empty ending with ',' or non-empty or empty quoted string */

Reply:
I rewrote the condition and added a comment, which is hopefully an improvement :) Fixed in [70241f1](https://github.com/wiredtiger/wiredtiger/pull/10267/commits/70241f1a1fcf8c5b60f1e9303b46e4568d643a27).